### PR TITLE
chore(sdk): Define `MaybeSerde`

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/paradigmxyz/reth/issues/12575

interfaces nicely with alloy types for which serde serialisation is feature gated

### CODE REVIEW

File: crates/primitives-traits/src/lib.rs [hunk 1]
```diff
@@ -92,3 +92,16 @@ pub trait MaybeArbitrary {}
 impl<T> MaybeArbitrary for T where T: for<'a> arbitrary::Arbitrary<'a> {}
 #[cfg(not(any(feature = "test-utils", feature = "arbitrary")))]
 impl<T> MaybeArbitrary for T {}
+
+/// Helper trait that requires de-/serialize implementation since `serde` feature is enabled.
+#[cfg(feature = "serde")]
+pub trait MaybeSerialize: serde::Serialize + for<'de> serde::Deserialize<'de> {}
```
Comment (mattsse) [inline crates/primitives-traits/src/lib.rs:98]: since this is ser + de perhaps we should rename this to MaybeSerde?

**Review by mattsse — COMMENTED** [review]
makes sense,
could you please double that reth-primitives compiles properly standalone, because on main `cargo t` fails rn which is kinda weird and I assume this is some weird issue with cfg_attr(test)

File: crates/primitives-traits/src/lib.rs (hunk 1 above)
Comment (emhane) [reply crates/primitives-traits/src/lib.rs:98]: this is the same on main, should be addressed in sep pr

Comment (emhane) [reply crates/primitives-traits/src/lib.rs:98]: https://github.com/paradigmxyz/reth/pull/12612

**Review by mattsse — APPROVED** [review]

**Pull request merged**